### PR TITLE
add user and group application

### PR DESCRIPTION
**Pull request by catalin-pitica-blankfactor**
- implement CRUD operations for group
- implement CRUD operation for user
- add error handling for invalid or not exist user or group
- integrate async content processing using BackgroundTasks

### CODE REVIEW

File: app/api/endpoints/group.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+from sqlalchemy.orm import Session
+from app.core.database import get_db
+from app.schemas.group_schema import (
+    GroupCreate,
+    GroupResponseForCreate,
+    GroupResponseForGet,
+)
+from app.service.group_service import GroupService
+
+router = APIRouter()
```
Comment (andrei-popescu-blankfactor) [inline app/api/endpoints/group.py]: You can probably move this entire module in api directly

File: app/api/endpoints/user.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+from app.service.group_service import GroupService
+from app.service.user_service import UserService
+from app.schemas.user_schema import (
+    UserCreate,
+    UserResponse,
+    UserResponseForGet,
+    UserUpdate,
+)
+
+router = APIRouter()
```
Comment (andrei-popescu-blankfactor) [inline app/api/endpoints/user.py]: Same as with group, I assume 🤷‍♂️

File: app/core/database.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+SQLALCHEMY_DB_URL = "sqlite:///./users.db"
```
Comment (andrei-popescu-blankfactor) [inline app/core/database.py]: I think you can add this to a config module

File: app/service/group_service.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+from enum import Enum
+from typing import Any
+from sqlalchemy.orm import Session
+from app.repository.group_repository import GroupRepository
+
+
+class GroupType(str, Enum):
```
Comment (andrei-popescu-blankfactor) [inline app/service/group_service.py]: This seems to be either a constant or a model?

**Review by andrei-popescu-blankfactor — COMMENTED** [review]
Great effort!!💥

File: app/service/group_service.py (hunk 4 above)
Comment (andrei-popescu-blankfactor) [reply app/service/group_service.py]: Can you please try to remove the `.idea` folder as well?

Comment (catalin-pitica-blankfactor) [reply app/service/group_service.py]: I defined this to be a constant

**Review by andrei-popescu-blankfactor — APPROVED** [review]
💪

**Pull request merged**